Fix Xcode backend

JohnCoates opened the pull request:
This makes the Xcode backend work.

* Stopped crashes from build.CustomTarget being unsupported.
* Fixes include directories.
* Now adds compiler arguments to targets
* Fixed typo causing Sources directory to show up as red.

I've been using this command to generate Xcode projects:
```
. build/frida-meson-env-macos-x86_64.rc; \
export builddir=build/xcode/frida-gum; \
mkdir -p $builddir; \
meson \
--prefix $PWD/frida/build/frida-macos-x86_64 \
--cross-file build/frida-macos-x86_64.txt \
--default-library static --backend xcode --buildtype minsize --strip -Denable_diet=auto \
frida-gum $builddir
```

JohnCoates committed:
added CustomTarget support to Xcode backend

- Fixed typo in Sources group sourceTree attribute

JohnCoates committed:
Use extra compiler aguments in Xcode backend

- Also fixes include paths by using absolute paths instead of the non-working relative ones

JohnCoates committed:
Correct error message in Xcode backend

Left in the error message when I copied this from the VS backend

oleavr commented:
Awesome!

Before landing this, let's remove the trailing whitespace and ensure that blank line separations are always just one blank line. (Unless the existing code uses e.g. two blank lines consistently in certain places, in which case it's fine.)

Would be awesome to submit this upstream after landing it.

JohnCoates committed:
Removed trailing whitespaces, silenced linter warnings

JohnCoates commented:
Okay I ran it through pylint and removed all the trailing whitespace warnings I saw, and removed the two blank lines I found.

Honestly I don't really know python, and getting this working was a struggle for me. Submitting upstream requires adding tests according to their guidelines, and it's too deep of a rabbit hole for me to jump into but of course anyone can feel free to do so.

Here are the issues that still crop up: 
* Crashes on pickling @ https://github.com/frida/meson/blob/master/mesonbuild/mesonmain.py#L186 - This is after the Xcode project file is generated, so it doesn't affect anything.
* Doesn't set include flag for `frida-version.h`
* Doesn't set Base SDK or architectures for iOS targets, so that has to be manually done for now
* I still had to manually add in a couple include directories when building frida-gumjs.

oleavr commented:
Totally understand! I'll merge this, and look into opening a PR upstream asking for volunteers to take it further. Even if they can't merge it as-is, someone might have been sitting on the fence thinking about fixing the Xcode backend – fingers crossed :-)

Thanks a lot for this awesome PR! ❤️
